lib/ansible/plugins/filter/aws_kms.py

```
@@ -0,0 +1,155 @@
+                role_session_name=basename(kwargs['role_to_assume'])
+            )
+        elif 'aws_access_key_id' in kwargs and 'aws_secret_access_key' in kwargs:
+            return boto3.Session(
+                aws_access_key_id=kwargs['aws_access_key_id'],
+                aws_secret_access_key=kwargs['aws_secret_access_key'],
+            )
+        return boto3.session.Session()
+    except Exception as ex:
+        raise AnsibleFilterError("Something went wrong while creating a "
+                                 "boto3 session in aws_kms_plugin - {0}".format(ex))
+
+
+def get_key_provider(key_arn, **kwargs):
+    """
+    Get KMS key provider object.
+    Args:
+      key_arn (str): AWS ARN to the KMS key.
+    Returns:
+      KMSMasterKeyProvider: KMS Master Key Provider object.
+    """
+
+    if not HAS_DEPENDENCIES:
+        raise AnsibleFilterError('You need to install "boto3" and "aws_encryption_sdk"'
+                                 'before using aws_kms filter')
+
+    if 'region_name' in kwargs:
+        region_name = kwargs['region']
+    else:
+        region_name = 'us-east-1'
+
+    boto_session = get_boto_session(**kwargs)
+    client = boto_session.client('kms', region_name=region_name)
+    key_provider = KMSMasterKeyProvider()
+    regional_master_key = KMSMasterKey(client=client, key_id=key_arn)
+    key_provider.add_master_key_provider(regional_master_key)
+    return key_provider
+
+
+def aws_kms_encrypt(plaintext, key_arn, **kwargs):
+    """
+    Encrypt with KMS.
+    Args:
+        plaintext (str): Plain text item to encrypt.
+        key_arn (str): AWS ARN to the KMS key.
+    Returns:
+        str: Encrypted item with KMS.
+    """
+    try:
+        ciphertext, dummy = encrypt(
+            source=plaintext,
+            key_provider=get_key_provider(key_arn=key_arn, **kwargs)
+        )
+        return base64.b64encode(ciphertext)
+    except AWSEncryptionSDKClientError as kms_ex:
```

When the dependency is not installed the error message isn't the one given on line 88 https://github.com/ansible/ansible/pull/42346/files#diff-8117f6964e26dcf20e852f21b50e2a70R88, but is a NameError since `AWSEncryptionSDKClientError` is undefined. The filter should fail for missing dependencies before using them.

---

@s-hertel @defionscode 

I've made these changes. Could you please have a look?